File: copilot_minimax_core/model_catalog.py
"""Builds JS snippets for model metadata injection."""

from __future__ import annotations

from .constants import MINIMAX_MODELS


def models_js() -> str:
    """Build the JS object literal for the hardcoded model catalogue."""
    entries = []
    for model_id, meta in MINIMAX_MODELS.items():
        bools = ",".join(
            f"{k}:{'!0' if v else '!1'}" if isinstance(v, bool) else f"{k}:{v}"
            for k, v in meta.items()
            if k != "name"
        )
        entries.append(f'"{model_id}":{{name:"{meta["name"]}",{bools}}}')
    return "{" + ",".join(entries) + "}"


def new_get_all_models_function() -> str:
    """Return replacement JS for provider model listing."""
    return (
        "async getAllModels(n,r){if(!r&&n)return[];try{"
        f"let a=Object.assign({models_js()},this._knownModels||{{}});"
        "return m9(this._name,a)}"
    )

File: packages/copilot-minimax-cli/src/modelCatalog.js
import { MINIMAX_MODELS } from './constants.js';

export function modelsJs() {
  const entries = Object.entries(MINIMAX_MODELS).map(([modelId, metadata]) => {
    const fields = Object.entries(metadata)
      .filter(([fieldName]) => fieldName !== 'name')
      .map(([fieldName, value]) => {
        if (typeof value === 'boolean') {
          return `${fieldName}:${value ? '!0' : '!1'}`;
        }
        return `${fieldName}:${value}`;
      })
      .join(',');
    return `"${modelId}":{name:"${metadata.name}",${fields}}`;
  });

  return `{${entries.join(',')}}`;
}

export function newGetAllModelsFunction() {
  return (
    'async getAllModels(n,r){if(!r&&n)return[];try{' +
    `let a=Object.assign(${modelsJs()},this._knownModels||{});` +
    'return m9(this._name,a)}'
  );
}
